mlir-clang/README.md

```diff
@@ -1,36 +1,42 @@
-##Build instructions
+## Build instructions
 
-        ## #Requirements -
-        working C and C++ toolchains(compiler, linker) - cmake - make
-    or ninja
+### Requirements 
+- Working C and C++ toolchains(compiler, linker)
+- cmake
+- make or ninja
 
-           ## #Process
-
-```
-#checkout the repository, 'cd' to it, then
-           mkdir
-           - p build cd build cmake../ llvm
-           - DLLVM_TARGETS_TO_BUILD =
-    "host;NVPTX" - DLLVM_ENABLE_PROJECTS =
-        "clang;mlir" - DLLVM_ENABLE_ASSERTIONS =
-            On - DCMAKE_BUILD_TYPE = Release - G Ninja #optional,
-             to use ninja instead of make make # or ninja; preferably with -j <num-cores>
+### 0. Clone Polygeist
+```sh
+git clone git@github.com:wsmoses/Polygeist.git $SOURCE_DIR
+cd $SOURCE_DIR
+git submodule init
+git submodule update
 ```
 
-Note that everything is statically linked, the produced binary
-should be freely movable around the file system.
-
-
-### How to run
-
+### 1. Install LLVM, MLIR, and Clang
+```sh
+mkdir $SOURCE_DIR/llvm-project/build
+cd $SOURCE_DIR/llvm-project/build
+cmake -G Ninja ../llvm \
+  -DLLVM_ENABLE_PROJECTS="mlir;clang" \
+  -DLLVM_TARGETS_TO_BUILD="X86" \
+  -DLLVM_ENABLE_ASSERTIONS=ON \
+  -DCMAKE_BUILD_TYPE=DEBUG
+ninja
+ninja check-mlir
 ```
-git clone https://github.com/Meinersbur/polybench
-cd polybench/polybench-code
-for i in `cat utilities/benchmark_list`;
-do
-  FNAME =`basename $i`;
-echo $FNAME;
-clang - E - Iutilities - DPOLYBENCH_USE_SCALAR_LB $i > / tmp / $FNAME;
-bin / mlir - clang / tmp / $FNAME main;
-done
+
+### 2. Install Polygeist
+```sh
+mkdir $SOURCE_DIR/build
+cd $SOURCE_DIR/build
+cmake -G Ninja .. \
+  -DMLIR_DIR=$PWD/../llvm-project/build/lib/cmake/mlir \
+  -DCLANG_DIR=$PWD/../llvm-project/build/lib/cmake/clang \
+  -DLLVM_ENABLE_ASSERTIONS=ON \
```

Comment:
I would remove the assertion/build type here, and also add the LLVM_EXTERNAL_LIT flag as here: https://github.com/wsmoses/Polygeist/blob/8fb9ac6429957b4c9206a775b2bb9c06e2799e18/.github/workflows/build.yml#L56

Reply:
LLVM_EXTERNAL_LIT should have a proper default if you're compiling against a build directory.  It should only be necessary if you're compiling against an install directory.

Reply:
Didn't know that! Yeah in which case so long as we're always building LLVM from scratch against the custom hash, no need for LLVM_EXTERNAL_LIT